server/index.ts

```diff
@@ -138,70 +122,72 @@ function setupGlobals() {
 	const { IPTools } = require('./ip-tools');
 	global.IPTools = IPTools;
 	void IPTools.loadHostsAndRanges();
-}
-setupGlobals();
 
-if (Config.crashguard) {
-	// graceful crash - allow current battles to finish before restarting
-	process.on('uncaughtException', (err: Error) => {
-		Monitor.crashlog(err, 'The main process');
-	});
+	const { TeamValidatorAsync } = require('./team-validator-async');
+	global.TeamValidatorAsync = TeamValidatorAsync;
 
-	process.on('unhandledRejection', err => {
-		Monitor.crashlog(err as any, 'A main process Promise');
-	});
+	global.Sockets = Sockets;
+	Sockets.start();
 }
 
-/*********************************************************
- * Start networking processes to be connected to
- *********************************************************/
+export const readyPromise = cleanupStale().then(() => {
+	setupGlobals();
+}).then(() => {
+	if (Config.usesqlite) {
+		require('./modlog').start();
+	}
 
-const { Sockets } = require('./sockets');
-global.Sockets = Sockets;
+	Rooms.global.start();
+	Verifier.start();
+	TeamValidatorAsync.start();
+	Chat.start();
 
-export function listen(port: number, bindAddress: string, workerCount: number) {
-	Sockets.listen(port, bindAddress, workerCount);
-}
+	/*********************************************************
+	 * Monitor config file and display diagnostics
+	 *********************************************************/
 
-if (require.main === module) {
-	// Launch the server directly when app.js is the main module. Otherwise,
-	// in the case of app.js being imported as a module (e.g. unit tests),
-	// postpone launching until app.listen() is called.
-	let port;
-	for (const arg of process.argv) {
-		if (/^[0-9]+$/.test(arg)) {
-			port = parseInt(arg);
-			break;
-		}
+	if (Config.watchconfig) {
+		ConfigLoader.watch();
 	}
-	Sockets.listen(port);
-}
 
-/*********************************************************
- * Set up our last global
- *********************************************************/
+	ConfigLoader.flushLog();
 
-const { TeamValidatorAsync } = require('./team-validator-async');
-global.TeamValidatorAsync = TeamValidatorAsync;
+	/*********************************************************
+	 * On error continue - enabled by default
+	 *********************************************************/
 
-/*********************************************************
- * Start up the REPL server
- *********************************************************/
+	if (Config.crashguard) {
+		// graceful crash - allow current battles to finish before restarting
+		process.on('uncaughtException', (err: Error) => {
+			Monitor.crashlog(err, 'The main process');
+		});
 
-// eslint-disable-next-line no-eval
-Repl.start('app', cmd => eval(cmd));
+		process.on('unhandledRejection', err => {
+			// TODO: rejections are not crashes,
+			// and unhandled rejections are not (always) bugs
```

Comment:
Is this true? I think unhandled rejections _are_ bugs.

Reply:
The dangling promises we leave everywhere aren't supposed to reject, though. They're very similar crashes.

I don't think it's a common pattern to add a handler after a Promise rejects. I think many runtimes treat an unhandled rejection very similarly to an unhandled exception and I think that's good.